File: Data Structures & Algorithms/longest-consecutive-sequence/submission-14.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) { 

        let set = new Set(nums)  
        let max = 0
         
        for (const num of set) {
            if (set.has(num - 1)) continue 

            let count = 1  
            let i = 1
            while(set.has(num + i)) {
                count ++  
                i ++
            } 
            max = Math.max(count, max)
        } 
        return max

    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-14.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = defaultdict(int) 
        res = 0 
        prefix = 0 
        count[0] = 1 

        for num in nums: 
            prefix += num 
             
            if prefix - k in count: 
                res += count[prefix - k] 

            count[prefix] += 1 

        return res